apm-sniffer/apm-sdk-plugin/spring-plugins/runner-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/runner/RunnerMethodInterceptor.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.spring.runner;
+
+import java.lang.reflect.Method;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+/**
+ * Record the Spring runner task local span.
+ */
+public class RunnerMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        String operationName = ComponentsDefine.SPRING_RUNNER.getName() + "/" + allArguments[0].getClass().getName();
+        AbstractSpan span = ContextManager.createLocalSpan(operationName);
+        Tags.LOGIC_ENDPOINT.set(span, Tags.VAL_LOCAL_SPAN_AS_LOGIC_ENDPOINT);
```

Comment:
Create a logic span in plugin? Why do we need to measure a local run? 

I have doubts about whether this is a general use case. If you want to measure this, the toolkit annotation and APIs could help you, rather than a public plugin.

Reply:
I have a SpringApplication with complex interactions with other services in CommandLineRunner. Skywalking show many segments at this situation and I try to integrate them into one segment according to code in `spring-scheduled-annotation-plugin`.

It's my fault to add Logic Endpoint here, the runner only run once.